src/models/GitWalkerFs.js

```diff
@@ -46,6 +46,7 @@ export class GitWalkerFs {
   }
 
   async readdir(entry) {
+    if ((await entry.type()) === 'linkTree') return []
```

Comment:
I think that it would be better to allow follow links in git walker but make it an option, in this case if someone wants old behavior he can still use it. Some users may rely on this feature so make the option `followSymlink: true` default.

Reply:
I think following symlink is never a "defined behavior" in git. Reading dir on such 'files' should work as other files. In fact, I think the signature of `readdir` should be changed to return both file name and file type, making use of Node's `withFileTypes` option instead of calling `stat` again and again.